group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -593,7 +595,8 @@ public List<DescribeGroupsResponseData.DescribedGroup> describeGroups(
      */
     ConsumerGroup getOrMaybeCreateConsumerGroup(
         String groupId,
-        boolean createIfNotExists
+        boolean createIfNotExists,
+        boolean addToGroupsMap
```

Comment:
I am not a fan of this new boolean flag because there is a change of misusing it. Have you considered keeping `getOrMaybeCreateConsumerGroup` on the `consumerGroupHeartbeat` path (without the part which adds to the map) and change the replay logic?